### PR TITLE
staveconnector (v5)

### PR DESCRIPTION
**Pull request by rvilarl**
StaveConnector & StaveText ported. Visual changes:

current
StaveConnector Combined_Draw_Test Petaluma svg_current
reference
StaveConnector Combined_Draw_Test Petaluma svg_reference

### CODE REVIEW

File: src/tables.ts [hunk 1]
```diff
@@ -79,8 +79,9 @@ export const CommonMetrics: Record<string, any> = {
   },
 
   StaveConnector: {
-    fontFamily: 'Times New Roman, serif',
-    fontSize: 16,
+    text: {
+      fontSize: 16,
+    },
```
Comment (ronyeh) [inline src/tables.ts:84]: I don't understand why this fontSize is under a text { } sub element, while below on line 109, the fontSize is declared directly on the StaveText element.

File: src/stavetext.ts [hunk 2]
```diff
@@ -29,79 +21,50 @@ export class StaveText extends StaveModifier {
...
+          y = stave.getYForBottomText(2);
         }
         break;
       default:
         throw new RuntimeError('InvalidPosition', 'Value Must be in Modifier.Position.');
     }
 
-    ctx.fillText('' + this.text, x, y + 4);
-    ctx.restore();
+    this.renderText(ctx, x, y + 4);
```
Comment (ronyeh) [inline src/stavetext.ts:67]: Do we know why this magic number +4 is here? If so, can we add a comment to explain it? :-)

File: src/staveconnector.ts [hunk 3]
```diff
@@ -255,9 +233,11 @@ export class StaveConnector extends Element {
...
         botY += 6;
         attachmentHeight = botY - topY;
-        Glyph.renderGlyph(ctx, topX - 5, topY, 40, 'bracketTop');
-        Glyph.renderGlyph(ctx, topX - 5, botY, 40, 'bracketBottom');
-        topX -= this.width + 2;
+        element.setText('\ue003' /* bracketTop */);
+        element.renderText(ctx, topX - 5, topY);
+        element.setText('\ue004' /* bracketBottom */);
+        element.renderText(ctx, topX - 5, botY);
+        topX -= 5;
```
Comment (ronyeh) [inline src/staveconnector.ts:240]: Can we explain (via comments) what these magic numbers -5 do? Also, why do we need to modify topX to shift it to the left by 5?

**Review by ronyeh — CHANGES_REQUESTED** [review]
Mostly looks good. I like how everything is now just Element (or collections of children Elements).

File: src/tables.ts (hunk 1 above)
Comment (rvilarl) [reply src/tables.ts:84]: StaveText is text only, as a consequence, no distinction is required. StaveConnector consist oh glyphs and text.

File: src/stavetext.ts (hunk 2 above)
Comment (rvilarl) [reply src/stavetext.ts:67]: I do not know. I did not look into that.

File: src/staveconnector.ts (hunk 3 above)
Comment (rvilarl) [reply src/staveconnector.ts:240]: I do not know. I did not look into that. There was two magic numbers 3 and 2 and I just replaced by 5.

File: src/stavetext.ts (hunk 2 above)
Comment (ronyeh) [reply src/stavetext.ts:67]: OK, later on we should play around with the magic numbers in the codebase to see what they all do, hahah. We can add comments or extract them into the Metrics file if they are something that should be customizable.

**Pull request merged**